src/generic/Matrix.jl

```diff
@@ -5015,13 +5015,11 @@ function (a::MatSpace{T})(b::Mat{T}) where {T <: RingElement}
    return b
 end
 
-function (a::MatSpace{T})(b::AbstractArray{T, 2}) where T <: RingElement
+function (a::MatSpace{T})(b::Array{T, 2}) where T <: RingElement
```

Comment:
Why this change?

Reply:
That's all the matrix constructor can accept anyway. And besides, the original issue was that this version was being called, when it certainly should not be.

Reply:
> That's all the matrix constructor can accept anyway

I think this change breaks the following:
```julia
julia> m = map(BigInt, sprand(Int8, 2, 3, .9))
2×3 SparseMatrixCSC{BigInt,Int64} with 6 stored entries:
  [1, 1]  =  -11
  [2, 1]  =  -8
  [1, 2]  =  -86
  [2, 2]  =  75
  [1, 3]  =  110
  [2, 3]  =  57

julia> a = MatrixSpace(ZZ, 2, 3)
Matrix Space of 2 rows and 3 columns over Integers

julia> a(m)
[-11  -86  110]
[ -8   75   57]

julia> typeof(ans)
AbstractAlgebra.Generic.MatSpaceElem{BigInt}

julia> @which a(m)
(a::AbstractAlgebra.Generic.MatSpace{T})(b::AbstractArray{T,2}) where T<:Union{RingElem, AbstractFloat, Integer, Rational} in AbstractAlgebra.Generic at /home/atma/.julia/packages/AbstractAlgebra/y3sVP/src/generic/Matrix.jl:5019
```

> the original issue was that this version was being called, when it certainly should not be.

I didn't carefully read the original issue, but a test would be useful to catch that bug, if we want to allow `AbstractArray` again in the future (or are the tests you included already catch that?)

Reply:
Works for me with the matrix_coerce branch. Are you sure it is broken?

Reply:
Nevermind, I guess this method is not called anymore, but there is a more general one right below which catches this example: `(a::MatSpace{T})(b::AbstractArray{S, 2}) where {S <: RingElement, T <: RingElement}`.

Reply:
That is certainly the intention.